Lesson16/main.js

```diff
@@ -0,0 +1,87 @@
+const tabContentList = document.getElementById("js-tab-content");
+const imgWrapper = document.getElementById("js-tab-img");
+const tabs = document.getElementsByClassName("tab-menu__item");
+
+
+async function getJsonOrError() {
+  const response = await fetch("./data.json");
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+async function fetchErrorHandling(response) {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+async function tryGetData() {
+  try {
+    const data = await getJsonOrError();
+    return data;
+  } catch (error) {
+    tabContentList.replaceWith(error);
+  }
+}
+
+async function createElement({ article }) {
+  const articleLength = 4;
+  const frag = document.createDocumentFragment();
+  for (let i = 0; i < articleLength; i++) {
+    const li = document.createElement("li");
+    const anchor = document.createElement("a");
+    anchor.href = "#";
+    anchor.insertAdjacentHTML("beforeend", article[i].title);
+    frag.appendChild(li).appendChild(anchor);
+  }
+  tabContentList.appendChild(frag);
+}
+
+function addImage({ image }) {
+  const img = document.createElement("img");
+  img.src = image;
+  imgWrapper.appendChild(img);
+}
+
+
+async function createTabContent(index) {
+  const data = await tryGetData();
+  if (data) {
+    createElement(data[index]);
+    addImage(data[index]);
+  }
+}
+
+
+async function init() {
+  const data = await tryGetData();
+  if (data) {
+    const hasSelectData = data.filter((value) => {
+      return value.select === true;
+    });
+    const tab = document.getElementById(hasSelectData[0].category);
```

Comment:
IMO
```suggestion
    const hasSelectData = data.find((value) => value);
    const tab = document.getElementById(hasSelectData.category);
```
> どのカテゴリタブを初期表示時に選んでいるかはデータとして持っている

So I think .find would be more appropriate since there is only one initial display

Reply:
Thanks for the review.
And thank you for sharing your cool way of writing using find.
I've fixed it.　
079c02c